homeassistant/components/fjaraskupan/fan.py

```diff
@@ -0,0 +1,197 @@
+
+    def _constructor(devicestate: DeviceState):
+        return [
+            Fan(devicestate.coordinator, devicestate.device, devicestate.device_info)
+        ]
+
+    await async_setup_entry_platform(
+        hass, config_entry, async_add_entities, _constructor
+    )
+
+
+class Fan(CoordinatorEntity[State], FanEntity):
+    """Fan device."""
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator[State],
+        device: Device,
+        device_info: DeviceInfo,
+    ) -> None:
+        """Init fan device."""
+        super().__init__(coordinator)
+        self._device = device
+        self._default_on_speed = 25
+        self._attr_name = device_info["name"]
+        self._attr_unique_id = device.address
+        self._attr_device_info = device_info
+        self._percentage = 0
+        self._preset_mode = PRESET_MODE_NORMAL
+        self._update_from_device_data(coordinator.data)
+
+    async def async_set_percentage(self, percentage: int) -> None:
+        """Set speed."""
+        new_speed = percentage_to_ordered_list_item(
+            ORDERED_NAMED_FAN_SPEEDS, percentage
+        )
+        await self._device.send_fan_speed(int(new_speed))
+        self.coordinator.async_set_updated_data(self._device.state)
+
+    async def async_turn_on(
+        self,
+        speed: str = None,
+        percentage: int = None,
+        preset_mode: str = None,
+        **kwargs,
+    ) -> None:
+        """Turn on the fan."""
+
+        if preset_mode is None:
+            preset_mode = self._preset_mode
+
+        if percentage is None:
+            percentage = self._default_on_speed
+
+        new_speed = percentage_to_ordered_list_item(
+            ORDERED_NAMED_FAN_SPEEDS, percentage
+        )
+
+        async with self._device:
+            if preset_mode != self._preset_mode:
+                if preset_mode == PRESET_MODE_AFTER_COOKING_MANUAL:
+                    await self._device.send_command(COMMAND_AFTERCOOKINGTIMERMANUAL)
+                elif preset_mode == PRESET_MODE_AFTER_COOKING_AUTO:
+                    await self._device.send_command(COMMAND_AFTERCOOKINGTIMERAUTO)
+                elif preset_mode == PRESET_MODE_NORMAL:
+                    await self._device.send_command(COMMAND_AFTERCOOKINGTIMEROFF)
+
+            if preset_mode == PRESET_MODE_NORMAL:
+                await self._device.send_fan_speed(int(new_speed))
+            elif preset_mode == PRESET_MODE_AFTER_COOKING_MANUAL:
+                await self._device.send_after_cooking(int(new_speed))
+            elif preset_mode == PRESET_MODE_AFTER_COOKING_AUTO:
+                await self._device.send_after_cooking(0)
+
+        self.coordinator.async_set_updated_data(self._device.state)
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new preset mode."""
+        if preset_mode == PRESET_MODE_AFTER_COOKING_MANUAL:
+            await self._device.send_command(COMMAND_AFTERCOOKINGTIMERMANUAL)
```

Comment:
It's easier to read if we move the calls out of the if/elif block and just store the command to send in a local variable here.

Reply:
Noticed i could switch to a dict lookup instead. Neater.